### PR TITLE
Improve feature sections responsive behavior

### LINKED ISSUE #308: Compress Feature page layout
The current feature page shows a lot of good stuff, but it requires a whole lot of scrolling. We could pretty easily put the images showcasing a feature to the left or right of the text to be able to fit a lot more stuff on the page.

https://amethyst.rs is a decent example of what I'm suggesting.

image

https://godotengine.org does it a bit differently, but it has multiple feature on each line so it's still more compressed than us.

image

### PR DESCRIPTION
**Pull request by doup**
- Improve features responsive behavior. Now it'll change the layout based on the width and orientation.
- Group classes under "feature" component
- Remove "reversed" classes in favor of `:nth-child(even)` selector

This is another preparation PR for #315 
Fixes #308

https://user-images.githubusercontent.com/188612/161916805-b31ab80b-ef63-4ab5-89bc-7f71a99b6a95.mp4

### CODE REVIEW

**Comment by rparrett** [comment]
I was initially confused about the "BEM naming," but it seems logical.

Tested in MacOS Chrome/Firefox and iOS Chrome/Safari and looks good to me.

**Comment by IceSentry** [comment]
I like it, it fixed the issue when I use a bigger default font-size!

File: sass/components/_feature.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
+.feature {
+    display: grid;
```
Comment (IceSentry) [inline sass/components/_feature.scss:2]: I completely forgot about using `display: grid` when I was trying to fix it. It's perfect for this.

**Review by IceSentry — APPROVED** [review]
LGTM

**Comment by doup** [comment]
@rparrett for a static site like this, it gives you poor man's encapsulation for the CSS (less chances to pollute & be polluted by other CSS classes; less specificity issues…) and the HTML is kind-of self documenting. The downside is its verbosity… I like that though. ^_^

**Comment by cart** [comment]
I do like that this is better at using available space, but I'd prefer it if we could maintain the aspect ratios of the image containers:

image

Not a huge fan of the "squashed" look.

**Comment by cart** [comment]
Or at least, I'd like to do a side by side comparison where it uses available space _and_ maintains aspect ratios. Its possible that maintaining aspect ratios _and_ using available space will feel "too big".

**Comment by cart** [comment]
I just noticed that the blender "whats new" feature overview is very similar to ours, and they manage to maintain aspect ratios, but they use a "wider" one than we currently do:
https://www.blender.org/download/releases/3-1/

**Comment by doup** [comment]
This one is tricky, making it fixed aspect-ratio in some cases (e.g. `2D Renderer`) makes it look too big. In Blender they are using the landscape aspect ratio for all the images; I'm not sure if it applies in this case. So instead, I've changed how the background color behaves, for mobile, instead of taking all the width it shrinks to the image.

I don't know, I'm still not convinced. Maybe is better to leave it as it was. 😓 

https://user-images.githubusercontent.com/188612/162588290-76b84bee-eceb-4d55-b62e-50090ecb09f0.mp4

**Review by SIGSTACKFAULT — APPROVED** [review]
Looks good but after a quick rebase it looks the same as main? Did this get rewritten and merged somewhere else?

**Comment by doup** [comment]
I've compared with `main` and doesn't seem to have a lot of conflicts… in any case, I've toyed a little bit with it and I'm still not convinced. The images appear with different sizes in some breakpoints… apart from other issues.

I might give it a try again (starting fresh from `main`), but for now I think I'll just close this PR. If anyone wants to give it a try, the interesting file is `sass/components/_feature.scss`.

**Pull request closed without merging**